llvm/lib/Target/AIE/AIEBasePipelinerLoopInfo.cpp

```diff
@@ -36,7 +36,45 @@ cl::opt<bool> TrackRegPressure(
 AIEBasePipelinerLoopInfo::AIEBasePipelinerLoopInfo(MachineInstr *EndLoop,
                                                    const AIEBaseInstrInfo &TII)
     : TII(TII), MRI(EndLoop->getMF()->getRegInfo()), EndLoop(EndLoop),
-      LoopBlock(EndLoop->getParent()) {}
+      LoopBlock(EndLoop->getParent()) {
+
+  const BasicBlock *BBLK = LoopBlock->getBasicBlock();
+  if (BBLK == nullptr)
+    return;
+
+  const Instruction *TI = BBLK->getTerminator();
+  if (TI == nullptr)
+    return;
+
+  MDNode *LoopID = TI->getMetadata(LLVMContext::MD_loop);
+  if (LoopID == nullptr)
+    return;
+
+  assert(LoopID->getNumOperands() > 0 && "requires at least one operand");
+  assert(LoopID->getOperand(0) == LoopID && "invalid loop");
+
+  for (unsigned i = 1, e = LoopID->getNumOperands(); i < e; ++i) {
+    MDNode *MD = dyn_cast<MDNode>(LoopID->getOperand(i));
+
+    if (MD == nullptr)
+      continue;
+
+    MDString *S = dyn_cast<MDString>(MD->getOperand(0));
+    if (S == nullptr)
+      continue;
+
+    if (S->getString() == "llvm.loop.itercount") {
+      assert(
+          (MD->getNumOperands() >= 2 && MD->getNumOperands() <= 3) &&
+          "Iteration count hint metadata should have two or three operands.");
```

Comment:
I feel that this message can be a bit confusing. Maybe: "one or two numerical operands".

Reply:
`MinTripCount` is  `int64_t`, is this way, I think it is better to use `getSExtValue`, insted.

Reply:
There's no way negative numbers make sense as an iteration count. 0 is fine as a "don't know" placeholder.

Reply:
oh, actually everything is signed. Sign extended as requested.